decidim-bulletin_board-js/src/client/message-parser.js

```diff
@@ -0,0 +1,99 @@
+import { JWS, JWK } from "node-jose";
+import {
+  MessageIdentifier,
+  AUTHORITY_TYPE,
+  BULLETIN_BOARD_TYPE,
+  TRUSTEE_TYPE,
+} from "./message-identifier";
+import { samePublicKeys } from "../utils";
+
+/**
+ * Verify and parses JWT tokens.
+ */
+export class MessageParser {
+  /**
+   * Initializes the class with the given params.
+   *
+   * @constructor
+   * @param {Object} params - An object that contains the initialization params.
+   * - {String} authorityPublicKeyJSON - The public key of the authority
+   */
+  constructor({ authorityPublicKeyJSON }) {
+    this.authorityPublicKeyJSON = authorityPublicKeyJSON;
+    JWK.asKey(authorityPublicKeyJSON, "json").then((result) => {
+      this.authorityPublicKey = result;
+    });
+    this.keys = null;
+  }
+
+  /**
+   * Parses the given token only if it can be verified.
+   *
+   * @param {String} token - A JWT token.
+   * @returns {Promise<Object>} - The payload included in the token.
+   * @throws An error is thrown if the payload is not a valid JSON or the token
+   *         cannot be verified.
+   */
+  async parse({ messageId, signedData }) {
```

Comment:
If you don't need the `messageId` I recommend just passing the `messageIdentifier` to the function to remove the dependency. What do you think?

Reply:
mm, I think that this is the best place to do it, as you always need to parse the `messageId` when you are parsing the `signedData`. If we move this to the caller, we would have to copy that line to the trustee and to the voter.